core/controlplane/config/templates/cloud-config-controller

```diff
@@ -968,6 +968,40 @@ write_files:
         sleep 3
       done
 
+      {{ if $.MigrateStacks -}}
+      list_not_empty() {
+        local file=$1
+        if ! [[ -s $file ]]; then
+          return 1
+        fi
+        if cat $file | grep -se 'items: \[\]'; then
+          return 1
+        fi
+        return 0
+      }
+
+      save_webhooks() {
+        local type=$1
+        local file=$2
+
+        echo "Storing and removing all ${type} webhooks to ${file}"
+        if [[ -s $file ]]; then
+          echo "$file already saved"
+        else
+          kubectl get ${type}webhookconfigurations -o yaml --export >$file
+          if list_not_empty $file; then
+            echo "deleting $type webhooks..."
+            ensuredelete $file
+          fi
+        fi
+      }
+
+      # Disable all mutating and validating webhooks because they can interfere with the stack migration)
+      echo "Storing and removing all validating and mutating webhooks..."
+      save_webhooks validating /srv/kubernetes/validating_webhooks.yaml
```

Comment:
So do we repeat these migration steps for each controller node being rolled, right?
Just asking for my understanding. Not thinking it is the cause of the slowness.

Reply:
Yes, I made it do this on each one in turn because I am worried about race conditions should all three controllers come up at the same time.